addons/stock/models/stock_inventory.py

```diff
@@ -350,6 +355,57 @@ def _run_inventory_tasks(self, company_id=False):
                              'company_id': location.company_id.id,
                              'location_ids': location})
 
+        # remove all previously generated 'is_conflict_inventory'
+        self.search([('state', '=', 'draft'), ('is_conflict_inventory', '=', True), ('create_date', '<', fields.Datetime.today())]).unlink()
+
+        # negative quantity check
+        quants = self._get_neg_qty_quants(company_id)
+        self._create_or_update_neg_inv(quants)
+
+    @api.model
+    def _get_neg_qty_quants(self, company_id=None, product_id=None):
+        domain = [('quantity', '<', 0.0), ('company_id', '!=', False), ('location_id.usage', 'in', ['internal', 'transit'])]
+        if company_id:
+            domain = expression.AND([[('company_id', '=', company_id)], domain])
+        if product_id:
+            domain = expression.AND([[('product_id', '=', product_id.id)], domain])
+        return self.env['stock.quant'].search(domain)
+
+    @api.model
+    def _create_or_update_neg_inv(self, quants, addOrRemove=None):
```

Comment:
Move this method in `stock.quant` model

Reply:
Done. Also moved the related SN method to `stock.production.lot`.

Unrelated, I also added in a test for each of the new features and made some of the names/method docstrings better.